### PR TITLE
Bump object to 0.19.0

### PR DESCRIPTION
**Pull request by bjorn3**
All changes: https://github.com/gimli-rs/object/compare/0.18.0...0.19.0

Changes affecting the writing part of object:

https://github.com/gimli-rs/object/commit/6ee7638493b14d682c3eb689725976b7d4be6552 fixes the symbol class for undefined symbols when writing COFF files. (required for Windows support in cg_clif)
https://github.com/gimli-rs/object/commit/47856a29c847467d925713066781b9164a096d1d and https://github.com/gimli-rs/object/commit/59a2cb6b521c46dc251ec3bfaf181b4a83759fc4 add basic support for AArch64 ELF file writing. (required for AArch64 support in cg_clif)
https://github.com/gimli-rs/object/commit/5fccab91fde14638db1af78847477e8361354a66 removes the `target-lexicon` dependency.

### CODE REVIEW

**Comment by yurydelendik** [comment]
@bjorn3 found this PR, and rebased/tweak it. Can you check if my changes are okay?

File: src/obj.rs [hunk 1]
```diff
@@ -13,6 +13,42 @@ use wasmtime_environ::{
...
+        architecture => {
+            bail!("target architecture {:?} is unsupported", architecture,);
+        }
+    };
+    let endian = match triple.endianness().unwrap() {
+        target_lexicon::Endianness::Little => object::Endianness::Little,
+        target_lexicon::Endianness::Big => object::Endianness::Big,
+    };
+    Ok((binary_format, architecture, endian))
+}
```
Comment (bjorn3) [inline src/obj.rs:50]: It would have been nice to deduplicate this, but I don't know of any suitable location.

Comment (yurydelendik) [reply src/obj.rs:50]: Yeah, I noticed that as well. I'm keeping in mind possibility of using cranelift-object in wasmtime -- but let's see how my further refactorings will go.

**Comment by bjorn3** [comment]
crates.io was having issues. It should work fine again.

**Pull request merged**